mongoengine/connection.py

```diff
@@ -1,17 +1,22 @@
 from pymongo import MongoClient, ReadPreference, uri_parser
-from mongoengine.python_support import (IS_PYMONGO_3, str_types)
+from mongoengine.python_support import (IS_PYMONGO_27, IS_PYMONGO_3, str_types)
 
 __all__ = ['ConnectionError', 'connect', 'register_connection',
            'DEFAULT_CONNECTION_NAME']
 
 
 DEFAULT_CONNECTION_NAME = 'default'
+AUTHENTICATION_MECHANISM = 'DEFAULT'
+
 if IS_PYMONGO_3:
     READ_PREFERENCE = ReadPreference.PRIMARY
 else:
     from pymongo import MongoReplicaSetClient
     READ_PREFERENCE = False
 
+if IS_PYMONGO_27:
```

Comment:
So in PyMongo v2.7 (https://github.com/mongodb/mongo-python-driver/blob/v2.7/pymongo/mongo_client.py#L372) I see:
```
mechanism = options.get('authmechanism', 'MONGODB-CR')
```

and in PyMongo v2.8+ I see:
```
mechanism = options.get('authmechanism', 'DEFAULT')
```

It seems that PyMongo v2.7 already defaults to `MONGODB-CR` and v2.8+ defaults to `DEFAULT` if `authmechanism` is not specified. I don't understand then why we have to explicitly specify it in MongoEngine... It seems that it should be enough to not pass it to PyMongo at all *unless* it was specified in the config. Am I missing something?